### PR TITLE
fix(FEC-8725): player stuck in loading state with deleted entries

### PR DESCRIPTION
**Pull request by odedhutzler**
Adding protection for cases the providers returns valid source object without the play sources

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

**Comment by kaltura-hooks** [comment]
Hi @odedhutzler,
Thank you for contributing this pull request!
Please sign the Kaltura CLA so we can review and merge your contribution.
Learn more at http://bit.ly/KalturaContrib

File: src/k-provider/common/base-provider.js [hunk 1]
```diff
@@ -66,6 +67,15 @@ export default class BaseProvider<MI> {
     throw new TypeError(`_parseDataFromResponse method must be implement by the derived class`);
   }
 
+  _verifyHasSources(sources: ProviderMediaSourcesObject) {
+    if ([...sources.hls, ...sources.dash, ...sources.progressive].length === 0) {
```
Comment (yairans) [inline src/k-provider/common/base-provider.js]: use [concat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat)

File: src/k-provider/common/base-provider.js [hunk 2]
```diff
@@ -66,6 +67,15 @@ export default class BaseProvider<MI> {
...
+      throw new Error(Error.Severity.CRITICAL, Error.Category.SERVICE, Error.Code.MISSING_PLAY_SOURCE, {
+        action: '',
+        messages: `No play source for entry id: ${typeof sources.id === 'string' ? sources.id : 'no entry id'}`
```
Comment (yairans) [inline src/k-provider/common/base-provider.js]:
```suggestion
        messages: `No play source for entry id: ${sources.id}`
```

File: src/k-provider/common/base-provider.js [hunk 3]
```diff
@@ -66,6 +67,15 @@ export default class BaseProvider<MI> {
     throw new TypeError(`_parseDataFromResponse method must be implement by the derived class`);
   }
 
+  _verifyHasSources(sources: ProviderMediaSourcesObject) {
+    if (sources.hls.concat(sources.dash).concat(sources.progressive).length === 0) {
```
Comment (OrenMe) [inline src/k-provider/common/base-provider.js]: concat accepts optional multi array params: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat

**Pull request merged**